src/main/java/com/derongan/minecraft/deeperworld/listeners/PlayerListener.kt

```diff
@@ -16,9 +17,9 @@ object PlayerListener : Listener {
     @EventHandler
     fun onPlayerTeleport(event: PlayerTeleportEvent) {
         val (player, _, to, cause) = event
-        if (player.gameMode == SURVIVAL
-            && (cause == ENDER_PEARL || cause == CHORUS_FRUIT)
-            && to?.section == null
+        if ((player.gameMode == SURVIVAL
+                && (cause == ENDER_PEARL || cause == CHORUS_FRUIT)
+                && to?.section == null) && player.canMoveSections
```

Comment:
Please follow the style of the other checks and add this one on a newline and without the brackets around all the previous ones. I also have a feeling you're using the old coding style which is what caused that extra spacing to be added in front of the `&&`s, see [here](https://github.com/MineInAbyss/MineInAbyss/wiki/Setup-and-Contribution-Guide#code-style-guide)